### PR TITLE
DEVTOOLS-76: Unchanged local file deleted remotely is a conflict

### PR DESCRIPTION
**Pull request by juancoen**
- When a locally unchanged file was deleted remotely, the app should
alert the user.

### CODE REVIEW

File: tests/unit/localService.test.js [hunk 1]
```diff
@@ -14,33 +14,61 @@ var fileSystemRepositoryStub = {};
...
   'api3.raml',
+  // [3] Changed locally // Deleted remotely
   'api4.raml',
+  // [4] Changed locally // Deleted remotely
+  'api5.raml',
+  // [5] Added locally
   'schema.json',
+  // [6] Changed locally
   'another.json',
+  // [7] Added locally // Already exists remotely
```
Comment (fschargorosdky) [inline tests/unit/localService.test.js]: you can change the names to reflect their behaviour in the test:
DeletedRemotely1.raml
ChangedRemotely1.raml
DeletedRemotely2.raml

That way you wont need comments

File: tests/unit/localService.test.js [hunk 2]
```diff
@@ -127,15 +160,18 @@ describe('localService', function () {
...
-          result.deleted.should.containEql(currentWorkspace.files[4].path);
-          result.deleted.should.containEql(currentWorkspace.files[5].path);
-          result.added.should.containEql(fileList[4]);
-          result.added.should.containEql(fileList[5]);
+          result.unchanged.should.containEql(fileList[0]); // api1.raml
+          result.unchanged.should.containEql(fileList[1]); // api2.raml
+          result.changed.should.containEql(fileList[2]); // api3.raml
+          result.changed.should.containEql(fileList[3]); // api4.raml
+          result.changed.should.containEql(fileList[4]); // api5.raml
+          result.changed.should.containEql(fileList[6]); // another.json
```
Comment (fschargorosdky) [inline tests/unit/localService.test.js]: you can use named references to reflect the objects behaviour
```js
var deletedRemotely1 = fileList[0]
```

File: tests/unit/localService.test.js (hunk 1 above)
Comment (juancoen) [inline tests/unit/localService.test.js]: As contentGenerator returns apiN.raml files for workspace, I think comments are ok

File: tests/unit/localService.test.js (hunk 2 above)
Comment (juancoen) [inline tests/unit/localService.test.js]: objects 'behaviour' can be inferred from the array in which they are placed. The idea of the comments is to match array positions to file names, which is useful when making changes to the test data.

File: tests/unit/localService.test.js (hunk 1 above)
Comment (fschargorosdky) [inline tests/unit/localService.test.js]: You can rename them.
If you use don't make they name reflect their behaviour it's very difficult to follow the tests. Which in turn makes them very hard to update.

Comment (juancoen) [inline tests/unit/localService.test.js]: renaming them one by one makes the generator useless.
And I think comments describing the files purpose are ok for a test.

Comment (juancoen) [inline tests/unit/localService.test.js]: done, used original file names but using representative variable names

File: tests/unit/localService.test.js (hunk 2 above)
Comment (juancoen) [inline tests/unit/localService.test.js]: done, used original file names but using representative variable names

File: tests/unit/localService.test.js [hunk 3]
```diff
@@ -114,12 +142,12 @@ describe('localService', function () {
   describe('status', run(function (localService) {
     it('should return the current local status', function (done) {
       fileSystemRepositoryStub.getFileHash.returns(BPromise.resolve(fileHash));
-      fileSystemRepositoryStub.getFileHash.onFirstCall().returns(
-        BPromise.resolve(currentWorkspace.files[0].hash));
-      localService.status()
+
+      localService.getStatus()
         .then(function (result) {
+          console.log(result);
```
Comment (fschargorosdky) [inline tests/unit/localService.test.js]: forever alone console

Comment (juancoen) [inline tests/unit/localService.test.js]: done

**Comment by fschargorosdky** [comment]
Looks good to me

**Pull request merged**